### PR TITLE
fix(ui): give feedback when a contact or channel import arrives while disconnected

### PR DESCRIPTION
**Pull request by jamesarich**
Scanning a shared-contact QR (`meshtastic.org/v/#…`) or channel-set QR (`meshtastic.org/e/#…`) while no device is connected gave zero feedback: the import request was parked in `UIViewModel` state, but the consuming dialogs in `SharedDialogs` only render while `ConnectionState.Connected`, so from the user's perspective the scan silently did nothing. This adds minimal snackbar feedback so the user knows the import was received and what to do next.

### 🐛 Bug Fixes
- `UIViewModel.setSharedContactRequested` / `setRequestChannelSet` now enqueue a snackbar via the existing `SnackbarManager` when an import arrives while the device is not connected: "Contact received. Connect to a device to import it." / "Channel settings received. Connect to a device to import them."
- Hooking the setters (rather than `handleDeepLink`) also covers the Discovery screen's join-offer path, which feeds the same gated dialogs.
- Import semantics are unchanged: the pending request still sits in state and the import dialog still appears automatically once the device connects. The URL format is untouched.

### Notes
- `SnackbarManager` buffers events in a `Channel(BUFFERED)` consumed only when a host collects, so a message enqueued before `MainScreen` composes (e.g. during app intro, or a cold-start deep link) is queued, not dropped.
- The snackbar can also fire while `Connecting`/`DeviceSleep`, where the dialog may pop moments later anyway; suppressing it there would reintroduce silence if the connection attempt fails, so it is deliberately kept simple.

### Testing Performed
- Full local baseline in this worktree: `spotlessApply spotlessCheck detekt assembleDebug test allTests` — all green.
- Verified `SnackbarManager`'s buffered-channel behavior covers the no-host-composed edge case (app intro screen).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added notifications for pending contact and channel imports when the device is disconnected.
  * Users are prompted to connect their device before importing shared contact or channel settings.
  * Imports continue automatically through the existing flow once connected.

### CODE REVIEW

File: core/ui/src/commonMain/kotlin/org/meshtastic/core/ui/viewmodel/UIViewModel.kt [hunk 1]
```diff
@@ -318,6 +320,16 @@ class UIViewModel(
...
+        if (contact != null) notifyImportPendingIfNotConnected(Res.string.import_pending_contact_connect)
+    }
+
+    /**
+     * The import dialogs in `SharedDialogs` only render while [ConnectionState.Connected], so a QR scanned while
+     * disconnected would otherwise queue silently with no feedback.
+     */
+    private fun notifyImportPendingIfNotConnected(messageRes: StringResource) {
+        if (connectionState.value is ConnectionState.Connected) return
+        safeLaunch(tag = "notifyImportPending") { snackbarManager.showSnackbar(message = getString(messageRes)) }
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply core/ui/src/commonMain/kotlin/org/meshtastic/core/ui/viewmodel/UIViewModel.kt]: Switched to `getStringSuspend(messageRes)` in 435166778. Note the call was already non-blocking (the file imports CMP's suspend `org.jetbrains.compose.resources.getString`, not the blocking `core.resources.getString` overload), but the suspend wrapper matches the repo convention and removes the ambiguity.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
